plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/SQLEditor.java

```diff
@@ -2292,6 +2277,35 @@ protected void doSetInput(IEditorInput editorInput) throws CoreException {
         editorImage = new Image(Display.getCurrent(), baseEditorImage, SWT.IMAGE_COPY);
     }
 
+    @NotNull
+    private Runnable getEditorInputInitializer(@NotNull IEditorInput editorInput) {
```

Comment:
Let's not make it return a `Runnable`. Instead, make it do what its `run` method does, and at the call site wrap it into lambda.

Reply:
I need Runnable for UIExecutionQueue.queueExec

Reply:
Right, but is it the method's responsibility to make you a `Runnable`? I don't think so